src/allowed_args.cpp

```diff
@@ -0,0 +1,247 @@
+// Copyright (c) 2017 The Bitcoin Classic developers
+// Distributed under the MIT software license, see the accompanying
```

Comment:
I'm not an expert here, but I generally prefer the MIT/BSD style license, because it's the most permissive. I thought the rest of the codebase was MIT licensed? I do see that some of the new files you've added are GPLv3, but I don't understand how that's compatible with the rest of the MIT licensed codebase.  I'd love to read more of an explanation though!